src/DurableTask.SqlServer/DTUtils.cs

```diff
@@ -196,6 +196,7 @@ public static bool HasPayload(HistoryEvent e)
             return historyEvent.EventType switch
             {
                 EventType.ExecutionStarted => ((ExecutionStartedEvent)historyEvent).Version,
+                EventType.SubOrchestrationInstanceCreated => ((SubOrchestrationInstanceCreatedEvent)historyEvent).Version,
```

Comment:
This is an unrelated fix that I happened to notice independently. I'm just adding it to this PR as a convenience since it's a small change.